src/com/maddyhome/idea/vim/ex/handler/CmdHandler.kt

```diff
@@ -0,0 +1,78 @@
+package com.maddyhome.idea.vim.ex.handler
+
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.editor.Editor
+import com.maddyhome.idea.vim.ex.CommandHandler.Flag.*
+import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.ex.*
+import com.maddyhome.idea.vim.ex.vimscript.VimScriptCommandHandler
+
+/**
+ * @author ecourant
+ */
+class CmdHandler : CommandHandler(
+        commands("com[mand]"),
+        flags(RANGE_FORBIDDEN, DONT_REOPEN, ARGUMENT_OPTIONAL)
+), VimScriptCommandHandler {
+    private val overridePrefix = "!"
+
+    override fun execute(cmd: ExCommand) {
+        this.addAlias(cmd)
+    }
+
+    override fun execute(editor: Editor, context: DataContext, cmd: ExCommand): Boolean {
+        if (cmd.argument.trim().isEmpty()) {
+            return this.listAlias(editor)
+        }
+
+        return this.addAlias(cmd)
+    }
+
+    private fun listAlias(editor: Editor): Boolean {
+        val lineSeparator = System.lineSeparator()
+        val allAliases = VimPlugin.getCommand().listAliases()
+        val aliases = allAliases.map { entry: Map.Entry<String, String> ->
+            "${entry.key.padEnd(20)} ${entry.value}"
+        }.sortedWith(String.CASE_INSENSITIVE_ORDER).joinToString(lineSeparator)
+        ExOutputModel.getInstance(editor).output("--- Aliases ---$lineSeparator$aliases")
+        return true
+    }
+
+    private fun addAlias(cmd: ExCommand): Boolean {
+        var argument = cmd.argument.trim()
+        val overrideAlias = argument.startsWith(this.overridePrefix)
+
+        // We want to trim off any "!" at the beginning of the arguments.
+        // This will also remove any extra spaces.
+        argument = argument.removePrefix(this.overridePrefix).trim()
+
+        // We want to get the first character sequence in the arguments.
+        // eg. command! Wq wq
+        // We want to extract the Wq only, and then just use the rest of
+        // the argument as the alias result.
+        val alias = argument.split(" ")[0]
+        argument = argument.removePrefix(alias).trim()
+
+        // User-aliases need to begin with an uppercase character.
+        if (!alias[0].isUpperCase()) {
+            return false
```

Comment:
Do not leave the user without understanding what is wrong. Use `com.maddyhome.idea.vim.VimPlugin#showMessage`

Reply:
I also added messages in other spots too, matching the messages returned from Vim for the same scenarios.